File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
         boolean[] visited = new boolean[computers.length];
        int counter = 0;

        for(int i=0; i< computers.length; i++){
            if(!visited[i]){
                dfs(computers,i,visited);
                counter++;
            }
        }

        return counter;
    }
    
    static void dfs(int[][] computers, int x, boolean[] visited){
        visited[x] = true;
        for(int i=0; i< computers.length;i++){
            if(!visited[i] && computers[x][i] == 1){
                dfs(computers, i, visited);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    const visited = new Array(n).fill(0)
    let count = 0;

    for (let i=0; i<visited.length; i++) {
        if (visited[i] === 1) continue;

        count += 1;
        dfs(computers, visited, i);
    }

    return count;
}

function dfs(map, visited, node) {
    visited[node] = 1;

    for (let i=0; i<map[node].length; i++) {
        if ( map[node][i] == 1 && visited[i] === 0) {
            dfs(map, visited, i);
        }
    }
}